### ISSUE TITLE
NRW vDOP is blank while listed as best imagery for at least some areas

### ISSUE THREAD

**Issue by matkoniecz** [osmlab/editor-layer-index#2273]
https://www.openstreetmap.org/edit?editor=id#map=19/51.18137/7.17874

![screen02](https://github.com/osmlab/editor-layer-index/assets/899988/92feffdc-b692-4416-8452-6d360a592f13)

**Comment by tordans** [comment]
See also https://github.com/osmlab/editor-layer-index/issues/2262

**Comment by crackwitz** [comment]
It's been completely blank for weeks because all the imagery is now in the proper "NRW Orthophoto". They removed "done" areas from vDOP as they worked on them.

***Today*** it's even the **default** choice. "NRW Liegenschaftsregister" used to be the default but it loads really slowly, so that shouldn't be the first choice either.

***"NRW Orthophoto"*** should be top choice. It loads quickly and it's up to date.

I wish iD would remember my choices and other settings across sessions...

Unless someone else is working on it, I'll swap the `"best"` properties of NRW vDOP and NRW Orthophoto, then open a PR.

**Comment by tordans** [comment]
Offtopic: 
> I wish iD would remember my choices and other settings across sessions...

FYI: If you use the stand alone version from the readme in GitHub you get proper url parses that store the selected imagery. You can then bookmark that.

**Comment by crackwitz** [comment]
Thank you for that hint. I gather that involves running it locally. I might try that. I'm already redirecting tile requests through a local nginx because the NRW WMS tells clients to *not cache at all*, which I find silly... i.e. I'm already somewhat into running things locally.

I just noticed that this issue here might have been created with the intention of perhaps *removing* the vDOP entry. I have no strong feelings on that either way.

**Comment by tordans** [comment]
> Thank you for that hint. I gather that involves running it locally. I might try that.

No, you can use https://ideditor-release.netlify.app/#background=nrw_ortho_wms&disable_features=boundaries&map=18.14/50.77696/6.08326 (from https://github.com/openstreetmap/id?tab=readme-ov-file#participate)

**Comment by tordans** [comment]
Back to the issue at hand: I think @pathmapper it would be best if you took over here since you introduced the data in #2022.

We have a few things fix here, IMO
- there was a "best" flag on a non-aerial imagery, which was already fixed in https://github.com/osmlab/editor-layer-index/pull/2277
- the current "best" flag is not a good pick, because it does not cover the area that is defined, so users see a blank screen
   This might be fixed by https://github.com/osmlab/editor-layer-index/pull/2288
- the naming of the layers does not allow users to understand what the differences are

I did no research in this area other than scanning the comments in https://github.com/osmlab/editor-layer-index/pull/2022#issuecomment-2006064942 and similar issues.

But I think we have to improve the situation soon by…
- if possible, change the area-definition to the area where images are actually present; your comment above could mean that this might be possible. But should the area change constantly, that would not work.
- we need to pick a "best" layer that shows data for all the area that is defined in the area-defintion. Everything else is just bad UX for users. It is great if there is possibly newer data, but that has to be a secondary step, selected by mappers that know about those things.

Given those constrains, what would you say is the best way forward, @pathmapper

**Comment by pathmapper** [comment]
I don't have much to add to https://github.com/osmlab/editor-layer-index/pull/2022#issuecomment-2006064942:

It all depends on how **best** is definied, and IMHO there is currently no such clear definition. So this should be resolved.

I always understood **best** as _most recent available data_, because currentness of data is crucial if one wants to trace things for OSM.

> if possible, change the area-definition to the area where images are actually present; your comment above could mean that this might be possible. But should the area change constantly, that would not work.

The area is changing constantly, e.g. for 
https://ideditor-release.netlify.app/#background=nrw_idop_wms&disable_features=boundaries&map=18.56/51.34159/6.33003

`nrw_idop_wms` is currently more recent than `nrw_otho_wms` ([link](https://kreis-viersen.github.io/quattromap/?settings=N4IgtgxiBcBMA0IIAsYgEYBsCGEDWIimAjGgHIBKA6gAQAyArgGYAu6AlpgCY0Bi2AJ3QBTQiAD2paBMwB9SrUasO3GgFlhLbF2wthAOzHiADlOJFY5ajXYARAPIAFPoJFHL08XIU2HzjVo6eoaIJh4ADAB0AKxEAMxo9gDKaurYxvrsBKEJ0kbGuVGxIJgALFa0tgAqANJG5Xmhxg1FAL5AA#17.42/51.341606/6.329416))

![image](https://github.com/osmlab/editor-layer-index/assets/20856381/2f880db0-edc6-41c0-9c8b-490d8e083433)

> we need to pick a "best" layer that shows data for all the area that is defined in the area-defintion. 

Do we really need to pick a "best" layer? For NRW we are having three different aerial sources, and it's constantly changing which source of them is the most recent for a given area.

Given the lack of a clear definition for what is "best", I currently won't assign "best" to any of the three sources.

**Comment by crackwitz** [comment]
I think that casual users, entering the editor, should never be faced with a blank background layer. That is just a bad user experience. And it's annoying to *everyone* using the editor.

Those vDOP and iDOP layers are "incomplete" and constantly changing. Even worse, the vDOP layer "lost" content over time (as it was done processing and incorporated into "NRW Orthophoto", the normal DOP), so if one saw imagery with the default behavior yesterday, and then opened the editor again today, and it was blank where it showed something yesterday, that is utterly bewildering.

Those should definitely not be the default picks.

They should remain though, with a description of what to expect of those layers, and what not to expect.

One could also "advertise" those layers in the description of the regular "NRW Orthophoto".

I did *not* expect my PR to be considered controversial.

**Comment by crackwitz** [comment]
> Given the lack of a clear definition for what is "best", I currently won't assign "best" to any of the three sources.

What happens if none of the NRW sources are tagged as "best"?

If there are no other "best" sources, we'd get Bing Maps because it's alphabetically the first.

If there *are* other "best" sources, as happens near the borders and sometimes a sizable distance "inland", then those take precedence, but they're blank in NRW.

A simple and generally beneficial change is to "tolerate" `dop` (NRW Orthophoto) as "best". It _is_ the best choice in general, considering the downsides of `vdop` (completely blank!) and `idop` (blurry, 50% coverage).

**Comment by pathmapper** [comment]
> If there are no other "best" sources, we'd get Bing Maps because it's alphabetically the first.

True, that's also not nice. But at least there's no layer advertised as `best` which is outdated for large areas.  

I don't have a strong opinion here, go with `dop` as `best` if you want.

Just keep in mind that it's expecteded that `idop`will get 100% coverage this year so you will have two sources covering 100% of NRW but which are each only the most recent for ~50% compared to the other one. 
(btw the "blurryness" is due to the fact that `idop` has 20 cm ground resolution compared to `dop` which has 10 cm)

Also new `vdop` will be available _soon_ which then will be ~2 years more recent than `dop`.

I really do like the comment from https://github.com/osmlab/editor-layer-index/issues/130#issuecomment-387130202: 

> `best` was a good enough hack when we really only had Bing, Mapbox, and whatever local imagery to choose from. We have so many more imagery to trace from now that we need something a bit more nuanced (this is a great problem to have!)

So it's time for something new, looks like https://github.com/osmlab/editor-layer-index/issues/130 is a good place for anyone who has some suggestions to join the discussion.

**Comment by matkoniecz** [comment]
> I always understood best as most recent available data, because currentness of data is crucial if one wants to trace things for OSM.

Resolution also matters (extreme example: daily updated satellite imagery with resolution of 100m is worse than aerial with resolution of 1cm updated 6 months ago).

Quality of georeferencing, colours, when it was taken (leafless imagery) may also matter.
